lua/entities/gmod_wire_expression2/core/egpobjects.lua

```diff
@@ -0,0 +1,493 @@
+--
+--	File for EGP Object handling in E2.
+--
+
+local function Update(self, this)
+	self.data.EGP.UpdatesNeeded[this] = true
+end
+
+---- Type defintion
+
+registerType("egpobject", "xeo", nil,
+	nil,
+	nil,
+	function(retval)
+		if retval == nil then return end
+		if not istable(retval) then error("Return value is neither nil nor a table, but a " .. type(retval) .. "!", 0) end
+	end,
+	function(v)
+		return not istable(v)
+	end
+)
+
+__e2setcost(2)
+
+registerOperator("ass", "xeo", "xeo", function(self, args)
+	local lhs, op2, scope = args[2], args[3], args[4]
+	local rhs = op2[1](self, op2)
+	if rhs == nil then return nil end
+	
+	local Scope = self.Scopes[scope]
+	local lookup = Scope.lookup
+	if not lookup then lookup = {} Scope.lookup = lookup end
+	if lookup[rhs] then lookup[rhs][lhs] = true else lookup[rhs] = {[lhs] = true} end
+
+	Scope[lhs] = rhs
+	Scope.vclk[lhs] = true
+	return rhs
+end)
+
+e2function number operator_is(egpobject egpo)
+	return (egpo ~= nil and isfunction(egpo.DataStreamInfo)) and 1 or 0
```

Comment:
This works but a metatable check might be better

Reply:
Sorry, I'm not sure how to go about that. Do you mind explaining what you mean?